packages/integrations/mdx/src/plugins.ts

```diff
@@ -0,0 +1,273 @@
+import type { MemberExpression, Literal } from 'estree';
+import type { MarkdownAstroData, AstroConfig } from 'astro';
+import type { Data, VFile } from 'vfile';
+import { visit as estreeVisit } from 'estree-util-visit';
+import { jsToTreeNode } from './utils.js';
+import { nodeTypes } from '@mdx-js/mdx';
+import type { PluggableList } from '@mdx-js/mdx/lib/core.js';
+import type { Options as MdxRollupPluginOptions } from '@mdx-js/rollup';
+import { bold, yellow } from 'kleur/colors';
+import rehypeRaw from 'rehype-raw';
+import remarkGfm from 'remark-gfm';
+import remarkSmartypants from 'remark-smartypants';
+import rehypeCollectHeadings from './rehype-collect-headings.js';
+import remarkPrism from './remark-prism.js';
+import remarkShiki from './remark-shiki.js';
+import { MdxOptions } from './index.js';
+
+export function recmaInjectImportMetaEnvPlugin({
+	importMetaEnv,
+}: {
+	importMetaEnv: Record<string, any>;
+}) {
+	return (tree: any) => {
+		estreeVisit(tree, (node) => {
+			if (node.type === 'MemberExpression') {
+				// attempt to get "import.meta.env" variable name
+				const envVarName = getImportMetaEnvVariableName(node as MemberExpression);
+				if (typeof envVarName === 'string') {
+					// clear object keys to replace with envVarLiteral
+					for (const key in node) {
+						delete (node as any)[key];
+					}
+					const envVarLiteral: Literal = {
+						type: 'Literal',
+						value: importMetaEnv[envVarName],
+						raw: JSON.stringify(importMetaEnv[envVarName]),
+					};
+					Object.assign(node, envVarLiteral);
+				}
+			}
+		});
+	};
+}
```

Comment:
I renamed `data-astro-utils.ts` to `plugins.ts`, so you can safely ignore the rest of the file! This function is the new part.